src/inventory/mod.rs

```diff
@@ -134,31 +144,37 @@ pub struct InventoryContext {
     pub has_inv_open: bool,
     pub player_inventory: Arc<RwLock<PlayerInventory>>,
     pub base_inventory: Arc<RwLock<BaseInventory>>,
+    mouse_position: Option<(f64, f64)>,
+    conn: Arc<RwLock<Option<Conn>>>,
```

Comment:
Another thing that would be cool is to make Conn internally mutable, so we don't need manual locking at all and massively reduce the risk of deadlocks overall. But if we want to change this we need to make sure that the additional internal locking doesn't cause significant performance penalty. Again, this should probably be discussed separately and is out of scope for this pr.